osu.Game/Screens/Select/Leaderboards/BeatmapLeaderboard.cs

```diff
@@ -204,10 +204,11 @@ void localScoresChanged(IRealmCollection<ScoreInfo> sender, ChangeSet? changes,
                 }
                 else if (filterMods)
                 {
-                    // otherwise find all the scores that have *any* of the currently selected mods (similar to how web applies mod filters)
+                    // otherwise find all the scores that have all of the currently selected mods (similar to how web applies mod filters)
                     // we're creating and using a string list representation of selected mods so that it can be translated into the DB query itself
                     var selectedMods = mods.Value.Select(m => m.Acronym);
-                    scores = scores.Where(s => s.Mods.Any(m => selectedMods.Contains(m.Acronym)));
+
+                    scores = scores.Where(s => s.Mods.Select(m => m.Acronym).SequenceEqual(selectedMods));
```

Comment:
Is there a guarantee that `s.Mods` will be in the same order as `selectedMods`? I don't think so, and whenever they aren't in the same order, this will fail.

You can either sort both lists alphabetically to ensure ordering, or use `HashSet`.

Reply:
Thanks for the quick reply, there shouldnt be any issues now.